components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/config/DestinationFetcherConfig.java

```diff
@@ -0,0 +1,68 @@
+package com.sap.cloud.cmp.ord.service.config;
+
+import org.springframework.beans.factory.annotation.Value;
+
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.SSLContext;
+import org.apache.http.ssl.SSLContexts;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.web.client.RestTemplate;
+
+import com.sap.cloud.cmp.ord.service.client.DestinationFetcherClient;
+
+@Configuration
+public class DestinationFetcherConfig {
+
+    @Value("${destination_fetcher.reload_url}")
+    private String reloadUrl;
+
+    @Value("${destination_fetcher.sensitive_data_url}")
+    private String sensitiveDataUrl;
+
+    @Value("${destination_fetcher.user_context_header:user_context}")
+    private String userContextHeader;
```

Comment:
Currently there are two ways of SaaS application consumption on behalf of their customers.

One is via user_context header. This is what you assume and use directly in the destination fetcher to extract tenant and other consumer information.

However, there is a second way of doing this. Instead of user_context header, the SaaS app can pass the token of the consumer directly. This way they do not need to parse anything about their user and just forward the token to us.

With the current implementation the second flow will not work.

My proposal would be to implement id_token security in the destination fetcher and just forward the id_token from here to it. This way you are benefiting from the ory oathkeeper hydrators chain that respects all kinds of security permutations and pass only the end product of the authn/authz.

This will be better in the future as well as new authn/authz flows will work out of the box.

In general it is not adviced to have authn/authz logic in components directly as those are one of the most complex flows in compass and are encapsulated in the hydrator component. If any component needs user information down the road should use the id_token claims.